packages/contract/lib/contract.js

```diff
@@ -20,13 +20,17 @@ if (typeof Web3 === "object" && Object.keys(Web3).length === 0) {
     var instance = this;
     var constructor = instance.constructor;
 
-    // Disambiguate between .at() and .new()
+    // Disambiguate between .at()/.deployed() & .new()
+    // in this case contract is a contract address string
     if (typeof contract === "string") {
-      var web3Instance = new constructor.web3.eth.Contract(constructor.abi);
-      web3Instance.options.address = contract;
-      contract = web3Instance;
+      const contractInstance = new constructor.web3.eth.Contract(
+        constructor.abi
+      );
+      contractInstance.options.address = contract;
+      if (constructor._json.networks[constructor.network_id])
```

Comment:
May want braces here to avoid future problems

Reply:
>Or would it be possible to only populate transactionHash when the address passed to at matches the address in the artifact, in which case you can just look up the transaction hash there?

This makes sense!

Reply:
Actually, thinking about this some more. `MyContract.at('0x...')` will *only* return with a `transactionHash` if it's been deployed with Truffle to the current network (via `network_id`) that you're connected/connecting to.

So for example, if you used Truffle to migrate `MyContract` to Ropsten, but then try to get an instance of `MyContract` that you deployed to Rinkeby using something _other_ than Truffle (i.e. Remix), that contract's information won't be stored in the artifacts, so if you call `MyContract.at()` to the address on Rinkeby, the `transactionHash` won't be available.

Reply:
But there could be multiple instances of a contract on the same network.  Suppose you used Truffle to deploy `MyContract` to Rinkeby, and then deployed another copy of `MyContract` to Rinkeby by a different means; say for instance that it was created as part of a transaction and you got the address from an event.  Using `MyContract.at()` on that second address shouldn't give you a contract with the `transactionHash` of the first.

Reply:
Good point!